test/controllers/SeriesDetailsControllerSpec.scala

```diff
@@ -98,5 +101,84 @@ class SeriesDetailsControllerSpec extends FrontEndTestHelper {
 
       wiremockServer.verify(postRequestedFor(urlEqualTo("/graphql")))
     }
+
+    "create a consignment when a valid form is submitted and the api response is successful" in {
+      val client = new GraphQLConfiguration(app.configuration).getClient[ac.Data, ac.Variables]()
+      val consignmentId = 1
+      val seriesId = 1
+      val consignmentResponse: ac.AddConsignment = new ac.AddConsignment(Some(consignmentId), seriesId, UUID.randomUUID())
+      val data: client.GraphqlData = client.GraphqlData(Some(ac.Data(consignmentResponse)), List())
+      val dataString: String = data.asJson.printWith(Printer(dropNullValues = false, ""))
+      wiremockServer.stubFor(post(urlEqualTo("/graphql"))
+        .willReturn(okJson(dataString)))
+
+      val controller = new SeriesDetailsController(getAuthorisedSecurityComponents(), new GraphQLConfiguration(app.configuration), getValidKeycloakConfiguration)
+      val seriesSubmit = controller.seriesSubmit().apply(FakeRequest().withFormUrlEncodedBody(("series", seriesId.toString)).withCSRFToken)
+      playStatus(seriesSubmit) mustBe SEE_OTHER
+      redirectLocation(seriesSubmit) must be(Some(s"/consignment/$consignmentId/transfer-agreement"))
+    }
+
+    "redirect to the error page when a valid form is submitted but there is an error from the api" in {
```

Comment:
Does this test still pass? Does this mean that we're still redirecting to error pages?

(Note to self: this is where it would be really useful for us to fix the GitHub build reporting.)

Reply:
No they don't, there were a few errors which I've fixed.

Reply:
Nitpicky, but the test name still says "redirect".

Reply:
I've changed this to not say redirect.